src/components/templates/product-table/index.tsx

```diff
@@ -0,0 +1,614 @@
+      copy.thumbnail = product.thumbnail
+    }
+
+    await createProductHook.mutateAsync(copy, {
+      onSuccess: ({ product }) => {
+        navigate(`/a/products/${product.id}`)
+        toaster("Created a new return reason", "success")
+      },
+      onError: (error) => {
+        toaster(getErrorMessage(error), "error")
+      },
+    })
+  }
+
+  const getActionablesForProduct = (product) => {
+    return [
+      {
+        label: "Edit",
+        onClick: () => navigate(`/a/products/${product.id}`),
+        icon: <EditIcon size={20} />,
+      },
+      {
+        label: product.status === "published" ? "Unpublish" : "Publish",
+        onClick: () => {
+          setUpdateProduct(product)
+          updateProductHook.mutate(
+            {
+              status: product.status === "published" ? "draft" : "published",
+            },
+            {
+              onSuccess: () => {
+                toaster(
+                  `Successfully ${
+                    product.status === "published" ? "unpublished" : "published"
+                  } product`,
+                  "success"
+                )
+                refetch()
+              },
+              onError: (err) => toaster(getErrorMessage(err), "error"),
+            }
+          )
+          // Medusa.products
+          //   .update(product.id, {
+          //     status: product.status === "published" ? "draft" : "published",
+          //   })
+          //   .then(() =>
+          //     toaster(
+          //       `Successfully ${
+          //         product.status === "published" ? "unpublished" : "published"
+          //       } product`,
+          //       "success"
+          //     )
+          //   )
+          //   .then(() => refetch())
+          //   .catch((err) => toaster(getErrorMessage(err), "error"))
+        },
+        icon:
+          product.status === "published" ? (
+            <UnpublishIcon size={20} />
+          ) : (
+            <PublishIcon size={20} />
+          ),
+      },
+      {
+        label: "Duplicate",
+        onClick: () => handleCopyProduct(product),
+        icon: <DuplicateIcon size={20} />,
+      },
+      {
+        label: "Delete",
+        variant: "danger",
+        onClick: () => setDeleteProduct(product),
+        icon: <TrashIcon size={20} />,
+      },
+    ]
+  }
+
+  return (
+    <div className="w-full h-full overflow-y-scroll">
+      {isLoading || isRefetching || !products ? (
+        <div className="w-full pt-2xlarge flex items-center justify-center">
+          <Spinner size={"large"} variant={"secondary"} />
+        </div>
+      ) : (
+        <>
+          <Table
+            filteringOptions={
+              <ProductsFilter
+                setStatusFilter={setStatusFilter}
+                statusFilter={statusFilter}
+                setCollectionFilter={setCollectionFilter}
+                collectionFilter={collectionFilter}
+                collections={collectionsList}
```

Comment:
what do you think about creating a `CollectionsFilter` component that you can use inside the `ProductsFilter` that would be responsible for fetching the collections and rendering them? It could take a prop like `onSelect` or `onFilter` which would update the filters state with the chosen collection(s)

Idea here is to have the `useAdminCollections()` hook isolated in the `CollectionsFilter` instead of having it at the top level.